### PR TITLE
Update opus to 1.5.2

### PR DESCRIPTION
**Pull request by softins**
**Short description of changes**

Update Opus library to the latest 1.5.2 release, plus a couple of fixes.

CHANGELOG: Update Opus library to v1.5.2

**Context: Fixes an issue?**

Existing Jamulus is using V1.4 of Opus. There are improvements to V1.5.2, which may enhance audio quality in some situations. It's just good to be on a current version. This was based on the distribution tarball for V1.5.2, not on a gitlab checkout, which would require an extra configuration step before use.

A couple of extra fixes were added, one a backport of a fix for a segfault, and two to silence compiler warnings.

**Does this change need documentation? What needs to be documented and how?**
Probably not, as it's just internal.

**Status of this Pull Request**
Ready for review. Tested on Linux server and Mac client. Hard to tell if the audio is slightly improved, but it's certainly not worse.

**What is missing until this pull request can be merged?**
Review. Testing on other platforms.

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

**Comment by ann0see** [comment]
Thanks.

Did you report the compiler errors to upstream too?

**Comment by softins** [comment]
> Thanks.
> 
> Did you report the compiler warnings to upstream too?

Not yet, I wanted to see if they had been fixed in any branches first. But I intend to.

File: Jamulus.pro
Comment (ann0see) [inline Jamulus.pro:1]: Compiling on Debian with Qt 5 gives the following warning:

```
WARNING: Failure to find: libs/opus/src/tansig_table.h
```

I believe, you'll need to remove tansig_table.h

**Review by ann0see — CHANGES_REQUESTED** [review]

**Comment by ann0see** [comment]
Otherwise, works on Debian 12 (client)

File: Jamulus.pro
Comment (softins) [reply Jamulus.pro:1]: Looks like that include file no longer exists in Opus 1.5.2, so I'll just remove it from `Jamulus.pro`

Comment (ann0see) [reply Jamulus.pro:1]: Thanks.

**Review by ann0see — APPROVED** [review]

**Comment by ann0see** [comment]
@pljones @gilgongo Could you please also test this?

**Comment by pljones** [comment]
Jack on Windows has decided it's going to crash for me at the moment (it goes through phases).  No audio on my Linux box (and, as Jack on Windows crashes, I can't use Jack Net to send the audio over to Windows, either).  Audio out on Windows is okay.

**Review by pljones — APPROVED** [review]
Didn't crash (JACK and ASIO on Windows).  ASIO audio out was fine.

**Pull request merged**